plugins/identity-shiro/src/test/java/org/opensearch/identity/shiro/DelegatingRestHandlerTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.shiro;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.core.common.bytes.BytesArray;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+public class DelegatingRestHandlerTests extends OpenSearchTestCase {
+
+    public void testDelegatingRestHandlerShouldActAsOriginal() throws Exception {
```

Comment:
Nice test. Can we also add maybe a negative case etc.?

Reply:
FYI The Security plugin uses the same classes: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/filter/DelegatingRestHandler.java

This is a special test that uses reflection to ensure that all methods from BaseRestHandler are delegated to: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/BaseRestHandler.java

If the BaseRestHandler ever changes without a corresponding change in DelegatingRestHandler, this test will fail which is to be expected. Any change to BaseRestHandler needs to be accompanied with a change in DelegatingRestHandler to ensure that the wrapped handler is delegated to for any overridden implementations.